### PR TITLE
fix: Absolute angle takes into account BodyComponent ancestors too

### PR DESCRIPTION
**Pull request by maurovanetti**
# Description
Currently, the implementation of the `absoluteAngle` getter in `PositionComponent` only takes into account `PositionComponent` ancestors. The implementation simply runs through the whole hierarchy summing up the relative angles.

This gives a wrong result with Forge2D because `BodyComponent`s have angles too: in the common situation where a `PositionComponent` (e.g. a `SpriteComponent`) is the child of a `BodyComponent`, the `BodyComponent`'s rotation is ignored when computing the `absoluteAngle`.

The fix simply introduces a new minimal interface `HasAngle` that both `PositionComponent` and `BodyComponent` implement (`PositionComponent` implements it automatically via `AngleProvider`), and reimplements `absoluteAngle` so that it considers all ancestors with an angle and not just the `PositionComponent` ones.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

### CODE REVIEW

**Comment by spydon** [comment]
Good idea! `HasAngle` should be named `ReadOnlyAngleProvider` though, to follow the new conventions set by `ReadOnlySizeProvider`.

**Comment by maurovanetti** [comment]
> Good idea! `HasAngle` should be named `ReadOnlyAngleProvider` though, to follow the new conventions set by `ReadOnlySizeProvider`.

Oops, I missed it. But it's got a lower-case O for some reason.

Updated.

**Comment by spydon** [comment]
> But it's got a lower-case O for some reason.

Since it hasn't been exported previously, do you mind renaming it to `ReadOnlySizeProvider` and same with `ReadOnlyAngleProvider`? :)

**Comment by maurovanetti** [comment]
There we go.

File: packages/flame/lib/effects.dart [hunk 1]
```diff
@@ -30,6 +30,7 @@ export 'src/effects/provider_interfaces.dart'
     show
         AnchorProvider,
         AngleProvider,
+        ReadOnlyAngleProvider,
```
Comment (spydon) [inline packages/flame/lib/effects.dart:33]: Can you export the `ReadOnlySizeProvider` too? 😇

**Review by spydon — APPROVED** [review]
Lgtm!

File: packages/flame/lib/effects.dart (hunk 1 above)
Comment (maurovanetti) [reply packages/flame/lib/effects.dart:33]: Symmetry… so satisfactory.

**Pull request merged**